### PR TITLE
etcdmain: add options for pprof

### PR DESCRIPTION
**Pull request by mitake**
This commit adds new options for activating profiling based on pprof
in etcd process.
 - -enable-pprof: boolean type option which activates profiling
 - -pprof-addr: string type option which specifies address for pprof

### CODE REVIEW

**Comment by gyuho** [comment]
@mitake  We need to document new flags here https://github.com/coreos/etcd/blob/master/etcdmain/help.go as well. Thanks

**Comment by gyuho** [comment]
And also here https://github.com/coreos/etcd/blob/master/Documentation/configuration.md.

File: etcdmain/config.go [hunk 1]
```diff
@@ -230,6 +234,10 @@ func NewConfig() *config {
 	fs.Var(&flags.DeprecatedFlag{Name: "peers"}, "peers", "DEPRECATED: Use -initial-cluster instead.")
 	fs.Var(&flags.DeprecatedFlag{Name: "peers-file"}, "peers-file", "DEPRECATED: Use -initial-cluster instead.")
 
+	// profiler
+	fs.BoolVar(&cfg.enablePprof, "enable-pprof", false, "Enable pprof profiler")
+	fs.StringVar(&cfg.pprofAddr, "pprof-addr", "localhost:6060", "address for pprof profiler")
```
Comment (xiang90) [inline etcdmain/config.go]: How about adding a handler to client url instead at https://github.com/coreos/etcd/blob/master/etcdserver/etcdhttp/client.go#L103?

**Comment by xiang90** [comment]
@mitake Kindly ping.

**Comment by mitake** [comment]
@xiang90 sorry for my late reply. I'll update this PR today.

**Comment by mitake** [comment]
@xiang90 updated, PTAL.

File: Documentation/configuration.md [hunk 2]
```diff
@@ -263,6 +263,12 @@ Follow the instructions when using these flags.
 + Print the version and exit.
 + default: false
 
+## Profiling flags
+
+### -enable-pprof
++ Enable pprof profiler, location is client URL + "/debug/pprof"
```
Comment (xiang90) [inline Documentation/configuration.md]: Enable runtime profiling data via HTTP server. Address is at `client URL + "/debug/pprof"`

File: etcdmain/config.go [hunk 3]
```diff
@@ -123,6 +123,9 @@ type config struct {
 	v3demo   bool
 	gRPCAddr string
 
+	// profiler
```
Comment (xiang90) [inline etcdmain/config.go]: this comment seems to be useless. Make it more specific or just remove it.

File: etcdmain/config.go [hunk 4]
```diff
@@ -230,6 +233,9 @@ func NewConfig() *config {
 	fs.Var(&flags.DeprecatedFlag{Name: "peers"}, "peers", "DEPRECATED: Use -initial-cluster instead.")
 	fs.Var(&flags.DeprecatedFlag{Name: "peers-file"}, "peers-file", "DEPRECATED: Use -initial-cluster instead.")
 
+	// profiler
```
Comment (xiang90) [inline etcdmain/config.go]: this comment is the same.

File: etcdmain/config.go [hunk 5]
```diff
@@ -230,6 +233,9 @@ func NewConfig() *config {
...
+	fs.BoolVar(&cfg.enablePprof, "enable-pprof", false, "Enable pprof profiler, location is client URL + \"/debug/pprof\"")
```
Comment (xiang90) [inline etcdmain/config.go]: also update the description of this flag as mentioned above.

File: etcdmain/help.go [hunk 6]
```diff
@@ -139,5 +139,9 @@ experimental flags:
 		enable experimental v3 demo API.
 	--experimental-gRPC-addr '127.0.0.1:2378'
 		gRPC address for experimental v3 demo API.
+
+profiling flags:
+	--enable-pprof 'false'
+		Enable pprof profiler, location is client URL + \"/debug/pprof\"
```
Comment (xiang90) [inline etcdmain/help.go]: same here.

**Comment by xiang90** [comment]
@mitake Please fix the CI failure. LGTM after addressing the issues mentioned above. Thanks a lot for the contribution!

**Comment by mitake** [comment]
@xiang90 thanks for your comment, I fixed the points you mentioned. But I missed the CI failure... will fix it soon after checking.

**Comment by mitake** [comment]
@xiang90 seems that go 1.4 doesn't provide `pprof.Trace`, I just skipped it for not breaking build with go 1.4. We need to consider how to support the profiler later.

**Comment by xiang90** [comment]
@mitake 

> I just skipped it for not breaking build with go 1.4. We need to consider how to support the profiler later.

That is fine. But can you add a todo? The simplest way is to add Trace when we decided to drop the support of go1.4. We will drop the support when go1.6 is release and stable.

**Comment by mitake** [comment]
@xiang90 OK, I added the todo.

**Comment by xiang90** [comment]
LGTM. Defer to @gyuho

**Comment by gyuho** [comment]
lgtm. Thanks @mitake

**Pull request merged**